.changeset/long-icons-pull.md

```diff
@@ -0,0 +1,37 @@
+---
+'@xstate/react': patch
+---
+
+Correct reference to the machine in `useInterpret` hook.
+Make sure a warning is shown in the console in the development mode when machine reference is updated during the hook lifecycle. This usually happens when machine options are dependent on external values and they're passed via `withConfig`.
```

Comment:
```suggestion
Fixed a regression with a development-only warning not being shown when a machine reference is updated during the hook lifecycle. This usually happens when machine options are dependent on external values and they're passed via `withConfig`.
```

I've checked that this was introduced correctly https://github.com/statelyai/xstate/pull/918/files but regressed in https://github.com/statelyai/xstate/pull/1472/files